### PR TITLE
add group labels and issues methods

### LINKED ISSUE #97: Get issues from Group of projects
Add the ability to get all issues from Group.

Doc Gitlab API : 
https://docs.gitlab.com/ee/api/issues.html#list-group-issues

### PR DESCRIPTION
**Pull request by WebDucer**
fixes #97

### CODE REVIEW

**Comment by WebDucer** [comment]
@casz @nmklotas Should I create a new PR for group issues, or can I add this tho PR?

**Comment by WebDucer** [comment]
Added list group issues endpoint

File: src/GitLabApiClient/IssuesClient.cs [hunk 1]
```diff
@@ -23,21 +24,48 @@ public sealed class IssuesClient
...
             _queryBuilder = queryBuilder;
-            _projectIssuesQueryBuilder = projectIssuesQueryBuilder;
             _projectIssueNotesQueryBuilder = projectIssueNotesQueryBuilder;
         }
 
+        /// <summary>
+        /// Retrieves issues.
+        /// By default retrieves opened issues from all users.
+        /// </summary>
+        /// <param name="projectId">The ID, path or <see cref="Project"/> of the project.</param>
```
[the remark this answers is not in the document]
Comment (WebDucer) [reply src/GitLabApiClient/IssuesClient.cs]: @casz Thanks for suggestions. I will refactor this next time.

I don't think, we should prevent from setting both. The most specific wins (ProjectId in this case).

File: src/GitLabApiClient/IssuesClient.cs [hunk 2]
```diff
@@ -23,21 +24,49 @@ public sealed class IssuesClient
...
+        /// <param name="groupId">The ID, path or <see cref="Group"/> of the group.</param>
+        /// <param name="options">Issues retrieval options.</param>
+        /// <returns>Issues satisfying options.</returns>
+        public async Task<IList<Issue>> GetAllAsync(ProjectId projectId = null, GroupId groupId = null,
+            Action<IssuesQueryOptions> options = null)
+        {
+            var queryOptions = new IssuesQueryOptions();
+            options?.Invoke(queryOptions);
+
+            string url = null;
```
[the remark this answers is not in the document]
Comment (WebDucer) [reply src/GitLabApiClient/IssuesClient.cs]: @casz Will be refactored with the next push

File: src/GitLabApiClient/IssuesClient.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (casz) [reply src/GitLabApiClient/IssuesClient.cs]: the most specific makes sense

[the remark this answers is not in the document]
Comment (casz) [reply src/GitLabApiClient/IssuesClient.cs]: Perfect example 👍

**Comment by casz** [comment]
Would you mind waiting on me merging this till I have fixed #100 I'd like to finish it this weekend :)

**Comment by WebDucer** [comment]
> Would you mind waiting on me merging this till I have fixed/merged #100 I'd like to finish it this weekend :)

No problem. I'm on vacations till 7th January 😉 . I will have a look on my GH account, but I have no test system till I'll be back.

**Comment by casz** [comment]
@WebDucer integration tests are failing, seems your using project id instead of group id helper :D

**Comment by WebDucer** [comment]
@casz Will check this week. I'll try to get the Docker test environment running on my macOS.

**Comment by casz** [comment]
All you need is docker to be installed.

The unit test will take care of doing the docker run command

**Comment by WebDucer** [comment]
@casz I had problems to get this running on Windows (in my job) 😄 . Now I will try this at home on Mac.

**Comment by casz** [comment]
As part of #100 I fixed the whole docker container setup. Not using a persisted volume and each time you run the unit test you get a fresh instance. So it should run a lot smoother. From my experience on PRs and builds on master the integrations works each time! 😍

**Comment by casz** [comment]
Please do report back if you can get it to work for both Windows and Mac.

You just have to run `dotnet test --verbosity normal`
The test suite waits for the gitlab container to report healthy status.
You can monitor this with `docker ps`

Windows 10 you need to use docker for desktop running in Linux mode.
Windows 2019 Server you need to enable experimental feature that allows to run LCOW (Linux Containers on Windows)
Mac/Linux only needs docker-ce edition.
You properly still use docker for desktop on mac.

**Comment by WebDucer** [comment]
@casz Fixed tests and integrated the current master. Switched form `.Be` to `.BeEquivalentTo` comparisson. Seems the `UrlEncode` behaves different on different platforms (e.g.: `:` is encoded as `%3A` or `%3a`).

**Comment by casz** [comment]
Thanks 🙏

**Comment by casz** [comment]
Did you get to test the unit tests on both Windows and Linux? 🤔

**Comment by WebDucer** [comment]
@casz No, only on macOS, have no Windows at home. Will try it tomorrow on Windows at office (first day after vacations).

**Comment by WebDucer** [comment]
@casz Tested on Windows. It works on my side. All tests are green.

**Comment by casz** [comment]
Great, I'll give it a quick review once my laptop is not bitlocked 😭

Soon one of these days I am throwing Linux on this laptop

**Comment by WebDucer** [comment]
@casz You are right, blaim on me 😉 The deprecated calls are fixed in tests.

**Comment by casz** [comment]
I liked the fact that I could see annotated issues right in the pull requests.

**Pull request merged**